python/res/enkf/res_config.py

```diff
@@ -76,60 +74,67 @@ def __init__(self, user_config_file=None, config=None, throw_on_error=True):
-                     ert_workflow_list,
-                     hook_manager,
-                     ert_templates,
-                     plot_config,
-                     ecl_config,
-                     ensemble_config,
-                     model_config,
-                     log_config,
-                     config_content]
-
-            c_ptr = None
-
-            if not self.errors or not throw_on_error:
-                for conf in configs:
-                    conf.convertToCReference(None)
-                c_ptr = self._alloc_full(config_dir, user_config_file, *configs)
+        elif user_config_file is not None:
 
+            parser = ConfigParser()
+
+            config_content = self._alloc_config_content(user_config_file, parser)
         else:
-            c_ptr = self._alloc_load(user_config_file)
+            raise ValueError("No config provided")
+
+
+
+        config_dir = config_content.getValue(ConfigKeys.CONFIG_DIRECTORY)
+
+        subst_config = SubstConfig(config_content=config_content)
+        site_config = SiteConfig(config_content=config_content)
+        rng_config = RNGConfig(config_content=config_content)
+        analysis_config = AnalysisConfig(config_content=config_content)
+        plot_config = PlotSettings(config_content=config_content)
+        ecl_config = EclConfig(config_content=config_content)
+        log_config = LogConfig(config_content=config_content)
+
+        ert_workflow_list = ErtWorkflowList(ert_workflow_list=subst_config.subst_list,
+                                            config_content=config_content)
+
+        hook_manager = HookManager(workflow_list=ert_workflow_list,
+                                   config_content=config_content)
+
+        ert_templates = ErtTemplates(parent_subst=subst_config.subst_list,
+                                     config_content=config_content)
+
+        ensemble_config = EnsembleConfig(config_content=config_content,
+                                         grid=ecl_config.getGrid(),
+                                         refcase=ecl_config.getRefcase())
+
+        model_config = ModelConfig(config_content=config_content,
+                                   data_root=config_dir,
+                                   joblist=site_config.get_installed_jobs(),
+                                   last_history_restart=ecl_config.getLastHistoryRestart(),
+                                   sched_file=ecl_config._get_sched_file(),
+                                   refcase=ecl_config.getRefcase())
+
+        configs = [
+            subst_config,
+            site_config,
+            rng_config,
+            analysis_config,
+            ert_workflow_list,
+            hook_manager,
+            ert_templates,
+            plot_config,
+            ecl_config,
+            ensemble_config,
+            model_config,
+            log_config,
+            config_content]
+
+        c_ptr = None
+
+        if not self.errors or not throw_on_error:
```

Comment:
I know you just copied this code from what was there before, bu since we are at it, we could make this if a bit more sensible.
If I understood it correctly, `self.errors` can be populated only by the `build_content` function.
So I would invert the `if` and move it up to line 84. ```
if self.errors and throw_on_error:
  raise ASuitableException('A sensible message')
```

To be fair, it is not entirely clear to me what is the expected behavior when `throw_on_error` is `False`. Does it create an invalid config object? Or?

Reply:
Yeah, I agree it could be moved up there. As to what "not throw on error" means, I am not sure. There is a lot of undefined behavior laying around.